src/pushprocessor.ts

```diff
@@ -433,8 +433,74 @@ export class PushProcessor {
         return PushProcessor.cachedGlobToRegex[glob];
     }
 
+    /**
+     * Parse a dotted key into parts following the escape rules.
+     *
+     * @param str - The key of the push rule condition: a dotted field.
+     * @returns The unescaped parts to fetch.
+     */
+    private partsFotDottedKey(str: string): string[] {
```

Comment:
I'd really like to see some unit tests for this specific function. It definitely has some edge cases and feels very testable... E.g. what if a string ends with \ or . or \. ?

Reply:
Talking to @andybalaam out of band and the best way to do this is to make it public, but put a warning in the comment saying that it isn't for public consumption (or something to that effect).

Reply:
Will do, thanks for the hint! 👍